src/VisualStudio/CSharp/Impl/ChangeSignature/CSharpChangeSignatureViewModelFactoryService.cs

```diff
@@ -55,5 +56,7 @@ public override SymbolDisplayPart[] GeneratePreviewDisplayParts(AddedParameterVi
         public override bool IsTypeNameValid(string typeName) => !SyntaxFactory.ParseTypeName(typeName, options: s_langVersionLatestParseOptions).ContainsDiagnostics;
 
         public override SyntaxNode GetTypeNode(string typeName) => SyntaxFactory.ParseTypeName(typeName);
+
+        public override bool IsVoidTypeSyntax(SyntaxNode node) => node is PredefinedTypeSyntax { Keyword.RawKind: (int)SyntaxKind.VoidKeyword };
```

Comment:
```suggestion
        public override bool IsVoidTypeSyntax(SyntaxNode node) => node is PredefinedTypeSyntax(kind: SyntaxKind.VoidKeyword);
```

Reply:
It's a positional pattern :-)

You can use them on types like records, or types that have suitable Deconstruct method.

Reply:
When I was making this PR I for whatever reason was very confident, that `SemanticModel.GetTypeInfo()` on pure `void` keyword would not resolve `System.Void` symbol. That much confident, that I didn't even check that. So I did this thing with `IsVoidTypeSyntax` to cover both `void` and `System.Void` type expressions. This suggested change didn't work on its own because `Deconstruct` is an extension method from an unimported namespace, so I had to open VS to fix it. Code fixes didn't suggest me to add necessary `using`, which is odd, so I had to manually guess what it can be. I did it, but then it became interesting to me why code fix didn't work, so I decided to go-to-def on `kind` variable. There I saw that it checks the kind of the whole node and is generic to all syntax nodes, so my thought was that the `void` case was broken. I launched VS hive and was very confused, that it didn't. Only then I launched it again but with a debugger attached to realise, that semantic model resolves `System.Void` type from `void` keyword, so all this complication with `IsVoidTypeSyntax` was not necessary all that time. You probably overlooked, that I check kind of inner `Keyword` property, not the node itself when reviewing, and it launched this sequence of events which lead to quite a bit simplification in the end. Interesting story)